src/main/java/net/fabricmc/loom/configuration/providers/minecraft/assets/MinecraftAssetsProvider.java

```diff
@@ -135,13 +163,33 @@ public static void provide(MinecraftProvider minecraftProvider, Project project)
 							throw new RuntimeException("Failed to download: " + assetName, e);
 						}
 
+						try {
+							if (localFileChecksum == null) {
+								checksumInfos.put(entry.getKey(), Files.asByteSource(file).hash(Hashing.sha1()).toString());
```

Comment:
You should be able to pull the hash from the AssetObject. Might need adding to that record.